src/Listeners/UpdateSyncedResource.php

```diff
@@ -112,11 +113,24 @@ protected function updateResourceInTenantDatabases($tenants, $event, $syncedAttr
                     $localModel->update($syncedAttributes);
                 } else {
                     // When creating, we use all columns, not just the synced ones.
-                    $localModel = $localModelClass::create($eventModel->getAttributes());
+                    $localModel = $localModelClass::create($this->getResourceCreationAttributes($eventModel));
                 }
 
                 event(new SyncedResourceChangedInForeignDatabase($localModel, $tenant));
             });
         });
     }
+
+    protected function getResourceCreationAttributes(Syncable $model): array
+    {
+        $attributes = $model->getAttributes();
+
+        if ($model->getResourceCreationAttributes()) {
+            // If function returned array is key-value, We assume default values are provided
+            // if array is plain values, fetch attributes from model
```

Comment:
Please use perfect grammar in comments that are part of the package's codebase. This can also be rewritten to be more clear.

Reply:
Resolving this review because I've rewritten the entire method.